apps/web/app/(app)/environments/[environmentId]/settings/api-keys/actions.ts

```diff
@@ -1,11 +1,30 @@
 "use server";
 
-import { deleteApiKey, createApiKey } from "@formbricks/lib/services/apiKey";
+import { authOptions } from "@/app/api/auth/[...nextauth]/authOptions";
+import { hasUserEnvironmentAccess } from "@/lib/api/apiHelper";
+import { deleteApiKey, createApiKey, canUserAccessApiKey } from "@formbricks/lib/services/apiKey";
 import { TApiKeyCreateInput } from "@formbricks/types/v1/apiKeys";
+import { Session, getServerSession } from "next-auth";
 
 export async function deleteApiKeyAction(id: string) {
-  return await deleteApiKey(id);
+  const session: Session | null = await getServerSession(authOptions);
+  const authorized = await canUserAccessApiKey(session, id);
```

Comment:
Please use `isAuthorized`. I think we should make it consistent from now on prefixing boolean variables with `is` :-)